WordPress/gutenberg
# DataViews: fix nested button tags on sidebar.

Fixes an issue noticed by @ntsekouras on https://github.com/WordPress/gutenberg/pull/55924#pullrequestreview-1724173833. The markup we produced contained a nested button tag, additionally, when we clicked the DropDown it also triggered the click event in the list item. Both issues are fixed in this PR.

## Testing 
- Verify there are not browser warnings related to nested button tags.
- Verify we can open the custom view menu without that view becoming the active one.
